datasets.py

```diff
@@ -702,4 +702,148 @@ def from_path(self, file_path):
         else:
             anns = [None, [0], [country], file_path, None]
 
-        return image, anns
\ No newline at end of file
+
+        return image, anns
+
+#Works on celebA image dataset of celebrities
+class CelebADataset(data.Dataset):
+    
+    def __init__(self, transform):
+        self.transform = transform
+        
+        # Where the images are located (doesn't need to exist, but can be helpful for other functions)
+        self.img_folder = 'img_align_celeba'
+
+        self.annotations_folder = 'Anno'
+
+        # List of all of the image ids
+        self.image_ids = []
+        #Adds the title of the image as its ID (e.g. 000006.jpg = 000006)
+        with open('Anno/identity_CelebA.txt') as f:
+            
+            for line in f:
+                stripped_line = line.strip()
+                stripped_line = stripped_line.split()
+                self.image_ids.append(stripped_line[0])
+        
+        print("done with ids")
```

Comment:
maybe helpful to include the progress type message to this print? so I see you also print done with categories, done with gender counting, etc. this print could be something like "done with ids (1 of 4 dataset setup steps)" or something